libglusterfs/src/gf-io-common.c

```diff
@@ -0,0 +1,568 @@
+/*
+  Copyright (c) 2021 Red Hat, Inc. <https://www.redhat.com>
+  This file is part of GlusterFS.
+
+  This file is licensed to you under your choice of the GNU Lesser
+  General Public License, version 3 or any later version (LGPLv3 or
+  later), or the GNU General Public License, version 2 (GPLv2), in all
+  cases as published by the Free Software Foundation.
+*/
+
+#include <signal.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+
+#include <urcu/uatomic.h>
+
+#include <glusterfs/gf-io-common.h>
+
+#define LG_MSG_IO_THREAD_BAD_PRIORITY_LVL(_res) GF_LOG_ERROR
```

Comment:
This is based on a macro construct to make writing log messages shorter and cleaner (hopefully). This is the macro that uses it:

```c
#define gf_io_log(_res, _msg, _args...) \
    gf_msg("io", _msg##_LVL(_res), -(_res), _msg, _msg##_FMT, ##_args)
```
The `_msg` argument is one of the constants defined in `glusterfs/libglusterfs-messages.h`, and the macro appends the suffixes `_LVL` and `_FMT` to get the log level of the message and the string format.

The `_LVL` macro receives one argument that represents the error code in negative form. This can be used to decide the log level based on the error code. This version of the patch doesn't use this feature currently, but it could be used, for example, to log the result of a `pthread_cond_timedwait()` call as LOG_DEBUG level when the error is -ETIMEDOUT, and as LOG_ERROR level in all other cases. This can be done this way:

```c
#define LG_MSG_PTHREAD_COND_TIMEDWAIT_FAILED_LVL(_res) \
    (((_res) == -ETIMEDOUT) ? LOG_DEBUG : LOG_ERROR)
```